app/services/EtmpRegimeService.scala

```diff
@@ -16,33 +16,140 @@
 
 package services
 
-import connectors.EtmpConnector
+import connectors.{EnrolmentStoreConnector, EtmpConnector}
 import javax.inject.Inject
-import play.api.libs.json.JsValue
-import uk.gov.hmrc.http.HeaderCarrier
+import models.{BusinessCustomerDetails, BusinessRegistrationDetails, EnrolmentVerifiers, EtmpRegistrationDetails}
+import play.api.Logger
+import uk.gov.hmrc.auth.core.retrieve.v2.Retrievals
+import uk.gov.hmrc.auth.core.{AffinityGroup, AuthConnector, AuthorisedFunctions, User}
+import uk.gov.hmrc.http.{HeaderCarrier, HttpResponse}
 import utils.AWRSFeatureSwitches
+import play.api.http.Status._
 
 import scala.concurrent.{ExecutionContext, Future}
+import scala.util.{Failure, Success, Try}
 
-class EtmpRegimeService @Inject()(etmpConnector: EtmpConnector) {
+class EtmpRegimeService @Inject()(etmpConnector: EtmpConnector,
+                                  val enrolmentStoreConnector: EnrolmentStoreConnector,
+                                  val authConnector: AuthConnector) extends AuthorisedFunctions {
 
-  def getRegimeRefNumber(safeId: String, regime: String)(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[Option[String]] = {
-    etmpConnector.awrsRegime(safeId, regime).map { response =>
-      (response.json \ "regimeIdentifiers").asOpt[List[JsValue]].flatMap { regimeIdentifiers =>
-        regimeIdentifiers.headOption.flatMap { regimeJs =>
-          (regimeJs \ "regimeRefNumber").asOpt[String]
-        }
+  def getEtmpBusinessDetails(safeId: String)(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[Option[EtmpRegistrationDetails]] = {
+    etmpConnector.awrsRegime(safeId).map { response =>
+      Try(EtmpRegistrationDetails.etmpReader.reads(response.json)) match {
+        case Success(value) => value.asOpt
+        case Failure(_)     => None
       }
     }
   }
 
-  def checkETMPApi(safeId: String, regime: String)(implicit ec: ExecutionContext, hc: HeaderCarrier): Future[Option[String]] = {
-    if (AWRSFeatureSwitches.regimeCheck().enabled) {
-      getRegimeRefNumber(safeId, regime) recover {
-        case _ => None
-      }
-    } else {
+  def handleDuplicateSubscription(etmpRegistrationDetails: EtmpRegistrationDetails,
+                                  businessCustomerDetails: BusinessCustomerDetails)
+                                 (implicit hc: HeaderCarrier, ec: ExecutionContext): Future[Option[EtmpRegistrationDetails]] = {
+    authorised(User).retrieve(Retrievals.affinityGroup){ affGroup =>
+      Future(getEtmpRegistrationDetails(affGroup, businessCustomerDetails, etmpRegistrationDetails))
+    } recover {
+      case _ =>
+        None
+    }
+  }
+
+  val AWRS_SERVICE_NAME = "HMRC-AWRS-ORG"
```

Comment:
GREY - Random val floating in the middle of code. This is only used in upsertEacdEnrolment so it can just rest inside that function.

Reply:
I'll move it to a private val at the top